platforms/software/dependency-management/src/main/java/org/gradle/internal/artifacts/configurations/AbstractRoleBasedConfigurationCreationRequest.java

```diff
@@ -64,14 +72,26 @@ public Configuration verifyExistingConfigurationUsage(Configuration conf) {
         ConfigurationInternal internalConf = (ConfigurationInternal) conf;
 
         if (!getRole().isUsageConsistentWithRole(conf)) {
-            warnAboutNeedToMutateUsage(internalConf);
-            if (internalConf.usageCanBeMutated()) {
-                internalConf.setAllowedUsageFromRole(getRole());
```

Comment:
I believe this is the only place we call `setAllowedUsageFromRole`. We can delete that from `DefaultConfiguration `now.

Reply:
https://github.com/gradle/gradle/pull/33196